### PR TITLE
Implement jsx-equals-spacing rule

### PR DESCRIPTION
**Pull request by bolatovumar**
Issue #103

Ported over the rule from eslint-plugin-react.
Included the exact same tests as in that plugin.

No automatic fixer because I don;t know how it works and never use it myself.
I reckon it would be really easy to add the fixer if someone wanted to do.

### CODE REVIEW

**Comment by adidahiya** [comment]
I strongly recommend adding a fixer here, it should be pretty simple. You just have to add a text replacement for " " -> "".

Can you look into getting CI status reported here?

**Comment by bolatovumar** [comment]
hey @adidahiya, I'm running into some weird issues with the fixer.

Here is what the test output looks like without the fixer applied:
no_fix

It looks like what you would expect to see.

Here is what the test output looks like with the fixer applied:
with_fix

To me it looks like the test should pass yet it fails. Any pointers here?

As per the CircleCI status, I did build my fork through the CircleCI's online interface. It only showed me the master branch, however. Not sure where I'm going wrong here.

**Comment by adidahiya** [comment]
@bolatovumar can you push your in-progress fixer code here so I can take a look?

**Comment by bolatovumar** [comment]
@adidahiya, sorry, it took me a while. Add my failing WIP fixers above.

The commands I'm running to test the fixers are (from the root of the repo after running `npm run build`):
`./node_modules/.bin/tslint -r ./build/rules/ --test ./test/rules/jsx-equals-spacing/always`
`./node_modules/.bin/tslint -r ./build/rules/ --test ./test/rules/jsx-equals-spacing/never`

In both cases it looks like the console output I'm getting looks something like this:

capture

Hope this is enough detail.

**Comment by IllusionMH** [comment]
@bolatovumar I've seen something similar when forgot to add empty line at the end of a `.fix` file.

File: test/rules/jsx-equals-spacing/always/test.tsx.fix [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<App />;
+<App foo />
+<App foo = "bar" />
+<App foo = {e => bar(e)} />
+<App {...props} />
+
+<App foo = {bar} />
+
```
Comment (IllusionMH) [inline test/rules/jsx-equals-spacing/always/test.tsx.fix]: Redundant line

File: test/rules/jsx-equals-spacing/always/test.tsx.fix [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+<App foo = {bar} />
+
```
Comment (IllusionMH) [inline test/rules/jsx-equals-spacing/always/test.tsx.fix]: Redundant line

File: test/rules/jsx-equals-spacing/always/test.tsx.fix [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+<App foo = {bar} />
+
+<App foo = {bar} bar = {baz} />;
```
Comment (IllusionMH) [inline test/rules/jsx-equals-spacing/always/test.tsx.fix]: Probably missing empty line after this one.

File: test/rules/jsx-equals-spacing/never/test.tsx.fix [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<App />
+<App foo />
+<App foo="bar" />
+<App {...props} />
+<App foo={e => bar(e)} />
+<App foo={bar} />
+<App foo={bar} />
+<App foo={bar} bar={baz} />
```
Comment (IllusionMH) [inline test/rules/jsx-equals-spacing/never/test.tsx.fix]: Probably missing empty line after this one.

File: test/rules/jsx-equals-spacing/never/test.tsx.fix [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<App />
+<App foo />
+<App foo="bar" />
+<App {...props} />
```
Comment (IllusionMH) [inline test/rules/jsx-equals-spacing/never/test.tsx.fix]: Missing empty line after this one.

File: test/rules/jsx-equals-spacing/never/test.tsx.fix [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+<App foo={e => bar(e)} />
+<App foo={bar} />
+<App foo={bar} />
```
Comment (IllusionMH) [inline test/rules/jsx-equals-spacing/never/test.tsx.fix]: Missing empty line after this one.

**Comment by bolatovumar** [comment]
@IllusionMH, alright, adding and removing line breaks seems to help

**Pull request merged**